### PR TITLE
sql: Allow Alter Column Drop Identity Command

### PR DESCRIPTION
**Pull request by delphamk**
Informs: #110010

This change adds a new command, 'ALTER TABLE
ALTER [COLUMN] DROP IDENTITY [ IF EXISTS ]',
which allows an ID column to drop the related
its Identity constraint and related sequence.
'IF EXISTS' is optional and will provide a notice
if the column is not an identity.

Release note (sql change): Identity Columns can
drop the Identity constraint by running 'ALTER
TABLE t ALTER COLUMN c DROP IDENTITY [ IF EXISTS ]'.

### CODE REVIEW

**Comment by andrew-delph** [comment]
I think this got closed by mistake when syncing my fork!

File: pkg/sql/drop_sequence.go [hunk 1]
```diff
@@ -190,7 +189,7 @@ func (p *planner) canRemoveOwnedSequencesImpl(
 		var firstDep *descpb.TableDescriptor_Reference
 		multipleIterationErr := seqDesc.ForeachDependedOnBy(func(dep *descpb.TableDescriptor_Reference) error {
 			if firstDep != nil {
-				return iterutil.StopIteration()
+				return errors.Newf("multiple iterations")
```
Comment (andrew-delph) [inline pkg/sql/drop_sequence.go:192]: If StopIteration is returned then the error will be nil.

This also fixes an error which can be tested with
```

statement ok
CREATE TABLE test1 (a int GENERATED ALWAYS AS IDENTITY (START WITH 10));

statement ok
CREATE TABLE test2(id INT PRIMARY KEY DEFAULT nextval('public.test1_a_seq'));

statement ok
SET use_declarative_schema_changer = off

# this should error
statement ok
drop table test1;

statement error descriptor is being dropped
INSERT INTO test2 DEFAULT VALUES;
```

**Review by annrpom — APPROVED** [review]
great job and thanks for your contribution! just fixed up some typos and added pgcodes to the `statement error`s

(edit: i am merging this now)

bors r+

**Comment by andrew-delph** [comment]
Thank you!

**Pull request merged**